common/src/main/java/org/apache/comet/NativeBase.java

```diff
@@ -46,15 +46,24 @@ public abstract class NativeBase {
 
   private static final String libraryToLoad = System.mapLibraryName(NATIVE_LIB_NAME);
   private static boolean loaded = false;
+  private static Throwable loadErr = null;
   private static final String searchPattern = "libcomet-";
 
   static {
-    if (!isLoaded()) {
+    try {
       load();
+    } catch (Throwable th) {
+      LOG.warn("Failed to load comet library", th);
+      // logging may not be initialized yet, so also write to stderr
+      System.err.println("Failed to load comet library: " + th.getMessage());
+      loadErr = th;
     }
   }
 
-  public static synchronized boolean isLoaded() {
+  public static synchronized boolean isLoaded() throws Throwable {
```

Comment:
This is called from `isCometEnabled` and we already had error handling here which could never be triggered due to `isLoaded` never being capable of throwing an exception.

Reply:
Correction: calling isLoaded could have resulted in the `NoClassDefFoundError` but this did not have details of root cause.